solr/core/src/java/org/apache/solr/api/ContainerPluginsRegistry.java

```diff
@@ -114,6 +118,16 @@ public synchronized ApiInfo getPlugin(String name) {
     return currentPlugins.get(name);
   }
 
+  static class PluginMetaHolder {
+    private final Map<String, Object> original;
+    private final PluginMeta meta;
```

Comment:
Does `PluginMeta` still need a separate `pathPrefix`? I think this could become a config property.

Reply:
It's a standard property , that is a part of the plugin definition

Reply:
It made sense when all plugins were request handlers, which is no longer true. In case of plugins that are not handlers a "standard" `pathPrefix` property doesn't make sense.

Reply:
`PluginMeta` is the information used by the framework. It's optional, so your plugin can choose to ignore it

Reply:
I know that I can ignore it - my point was that this property is a relic of the time when we allowed only Api handlers as plugins. Now when non-Api plugins are first-class citizens half of the time this property doesn't make sense because it's specific only to Api plugins - so it should not be exposed as a standard property for all plugins.

Reply:
It can be a specific sub-class of `PluginMeta` say `EndPointPluginMeta` .

However, that change does not belong here

Reply:
Let's make it a sub-class.

(I would argue that it belongs in this PR because it's a part of the configuration mechanism for the plugins that this PR defines. Since we are adding a flexible config bean it doesn't make sense to still keep the old primitive field. And you can always ADD fields in subclasses but removing them is much harder... ;) )